### PR TITLE
Suggest new projection/install tree for spack to avoid collisions

### PR DESCRIPTION
**Pull request by adrienbernede**
I just ran into this:

```
==> Error: Install prefix collision for blt-0.4.1-bnajcsvekqqmr7pub7qx4iwshjni2iib
Prefix directory /Users/adrien/Work/Repos/serac_libs/apple-clang-13.1.6/blt-0.4.1 already used by another installed spec.
[ERROR: failure of spack install/dev-build]
```

Which confirms my opinion that the install tree should be more strict.

Note: with spack@0.18.0 adding full hash everywhere and in particular hashing the packages.py, it will be easier to have collisions.

- [ ] In CI this may very well cause a lot of new installations, not sure about that though -> Requires to run the full_workflow on Gitlab.
- [ ] See if what would be the consequence regarding the already installed dependencies if we need to update them.

### CODE REVIEW

File: scripts/spack/configs/config.yaml [hunk 1]
```diff
@@ -19,7 +19,7 @@ config:
   install_tree:
     root: $spack/..
     projections:
-     all: "${COMPILERNAME}-${COMPILERVER}/${PACKAGE}-${VERSION}"
+      all: "${ARCHITECTURE}/${COMPILERNAME}-${COMPILERVER}/${PACKAGE}-${VERSION}-${HASH}"
```
Comment (white238) [inline scripts/spack/configs/config.yaml]: `ARCHITECTURE` is handled by the build scripts and this will duplicate that

Comment (adrienbernede) [reply scripts/spack/configs/config.yaml]: Removed

**Review by white238 — APPROVED** [review]

**Review by samuelpmishLLNL — APPROVED** [review]

**Review by jamiebramwell — APPROVED** [review]

**Comment by adrienbernede** [comment]
I'll take those approvals as a reminder of the check I want to do before merging. ;)

**Comment by white238** [comment]
> I'll take those approvals as a reminder of the check I want to do before merging. ;)

I actually just tested it on both our rz configs and there was no difference other than the hash's now slowing up in the host-configs.  Which like makes me slightly sad due to clutter, but I suppose I can live with it. It will be rough.

**Comment by white238** [comment]
Feel free to merge.  I am going to do a rebuild of the TPLs very soon.

**Comment by adrienbernede** [comment]
OK then ;)

**Pull request merged**